mangadex_downloader/chapter.py

```diff
@@ -480,6 +496,14 @@ def _check_range_chapter_legacy(self, chap):
                 log.debug(f"Ignoring chapter {num_chap}, because chapter {num_chap} is in ignored list")
                 return False
 
+        if is_vol_number and num_vol > 0.0:
+            if self.start_volume is not None and not (num_vol >= self.start_volume):
+                log.debug(f"Ignoring chapter in volume {num_vol}, because volume {num_vol} is in ignored list")
+                return False
+            if self.end_volume is not None and not (num_vol <= self.end_volume):
+                log.debug(f"Ignoring chapter in volume {num_vol}, because volume {num_vol} is in ignored list")
+                return False
+
```

Comment:
This volume checking must be placed before chapter checking, if both start/end chapter and volume is used then `--start-volume` and `--end-volume` won't be useful enough.

For example:
```
mangadex-dl "https://mangadex.org/title/a96676e5-8ae2-425e-b549-7f15dd34a6d8/komi-san-wa-komyushou-desu" --start-chapter 1 --end-chapter 43 --start-volume 1 --end-volume 2
```

The user wants to stop download at volume number 2 but because of `--end-chapter` is set to 43, then it will continue to download until chapter 43, **not until volume 2**. Which you can see in provided MangaDex link above, volume 2 stopped at chapter 34

Overall, this PR looks good, i will try test and merge it once the fix is resolved 👍

Reply:
If you are confused, it should be like this !

![image](https://github.com/user-attachments/assets/f73e2b17-e715-45f6-8316-f1256dc0fe97)

I'm sorry i cannot use git diff in Github 🗿, so i use text comparer in devtoys